pylint/checkers/typecheck.py

```diff
@@ -1886,6 +1894,16 @@ def visit_with(self, node: nodes.With) -> None:
                         inferred.parent, self.linter.config.contextmanager_decorators
                     ):
                         continue
+                    # Check if it's an AsyncGenerator decorated with asynccontextmanager
+                    if isinstance(inferred, astroid.bases.AsyncGenerator):
+                        async_decorators = ["contextlib.asynccontextmanager"]
+                        if decorated_with(inferred.parent, async_decorators):
+                            self.add_message(
+                                "async-context-manager-with-regular-with",
+                                node=node,
+                                args=(inferred.parent.name,),
```

Comment:
```suggestion
                                args=(inferred.parent.name,),
                                confidence=INFERRED
```
Not the right import, but you get the idea, you'll also have to regenerate the expected output.